engine/src/srvwindows.cpp

```diff
@@ -681,11 +681,7 @@ struct MCWindowsSystem: public MCSystemInterface
 		if (t_long_fname == nil)
 			ep . setsvalue(fname);
 		else
-		{
-			delete fname;
-			fname = t_long_fname;
-			ep . setsvalue(fname);
-		}
+			ep . setsvalue(t_long_fname);
 		
 		if (t_ptr != NULL)
```

Comment:
You need an ep.grabsvalue() here to make sure the EP owns its contents. Otherwise if the EP isn't mutated, its contents will be freed by the subsequent deletes.